src/libPMacc/include/algorithms/math/doubleMath/floatingPoint.tpp

```diff
@@ -60,6 +72,29 @@ struct Float2int_rd<double>
     }
 };
 
+template<>
+struct Float2int_rn<double>
+{
+    typedef int result;
+
+    HDINLINE result operator( )(double value)
+    {
+#if __CUDA_ARCH__
+        return ::__double2int_rn( value );
+#else
+        if(value < 0.0)
+            return -(*this)(-value);
+        static const result epsilon = 1e-7;
+        /* Round towards nearest with .5 -> +inf */
```

Comment:
A static_cast<int>(x + 0.5) is the common way for rounding x to the nearest int and **THE** example for writing your own `round`-function.
The *.5 -> +inf* part means that when x = n + 0.5 (n=int) then x is rounded towards + infinity.